GenericsHomework/Node.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System.Collections;
+
+namespace GenericsHomework;
+[System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "CA1710:IdentifiersShouldHaveCorrectSuffix", Justification = "Benjamin is allowing this suppression.")]
+public class Node<T> : ICollection<T>
+{
+    private readonly T _value;
+    private Node<T> _next;
+    public Node(T value)
+    {
+        _value = value;
+        _next = this;
+    }
+
+    public T Value { get {  return _value; } }
```

Comment:
```suggestion
    public T Value { get; }
```
An auto property is preferred to be used here